src/gc/unix/cgroup.cpp

```diff
@@ -0,0 +1,271 @@
+            return false;
+    
+        strcpy(memsw_limit_filename, _memory_cgroup_path);
+        strcat(memsw_limit_filename, MEM_SWAP_LIMIT_FILENAME);
+        return ReadMemoryValueFromFile(memsw_limit_filename, val);
+    }
+private:
+    bool FindMemoryHierarchyMount(char* mountpath)
+    {
+        bool retval = false;
+        char *line = nullptr;
+        size_t lineLen = 0;
+        FILE *mountinfofile = fopen(PROC_MOUNTINFO_FILENAME, "r");
+        if (mountinfofile == NULL)
+            goto done;
+    
+        while (!retval && getline(&line, &lineLen, mountinfofile) != -1)
+        {
+            char filesystemType[100];
+            char options[MAX_PATH];
+    
+            char* separatorChar = strchr(line, '-');
+    
+            // See man page of proc to get format for /proc/self/mountinfo file
+            int sscanfRet = sscanf(separatorChar, 
+                                   "- %s %*s %s",
+                                   filesystemType,
+                                   options);
+            if (sscanfRet != 2)
+            {
+                assert(!"Failed to parse mount info file contents with sscanf.");
+                goto done;
+            }
+    
+            if (strncmp(filesystemType, "cgroup", 6) == 0)
+            {
+                char* context = NULL;
+                char* strTok = strtok_r(options, ",", &context); 
+                while (strTok != NULL)
+                {
+                    if (strncmp("memory", strTok, 6) == 0)
+                    {
+                        sscanfRet = sscanf(line,
+                                           "%*d %*d %*s %*s %s ",
+                                           mountpath);
+                        if (sscanfRet != 1)
+                        {
+                            assert(!"Failed to parse mount info file contents with sscanf.");
+                            goto done;
+                        }
+                        retval = true;
+                        break;
+                    }
+                    strTok = strtok_r(NULL, ",", &context);
+                }
+            }
+        }
+    done:
+        if (line)
+            free(line);
+        if (mountinfofile)
+            fclose(mountinfofile);
+        return retval;
+    }
+    
+    bool FindCGroupPathForMemorySubsystem(char* cgrouppath)
+    {
+        bool retval = false;
+        char *line = nullptr;
+        size_t lineLen = 0;
+        FILE *cgroupfile = fopen(PROC_CGROUP_FILENAME, "r");
+        if (cgroupfile == NULL)
+            goto done;
+    
+        while (!retval && getline(&line, &lineLen, cgroupfile) != -1)
+        {
+            char subsystem_list[100];
+            char cgroup_path[MAX_PATH];
+    
+            // See man page of proc to get format for /proc/self/cgroup file
+            int sscanfRet = sscanf(line, 
+                                   "%*[^:]:%[^:]:%s",
+                                   subsystem_list,
```

Comment:
There are no secure versions of these in linux. Are you suggesting to not to use these non-secure crt functions.